tools/mate-session-check-accelerated-gles-helper.c

```diff
@@ -0,0 +1,201 @@
+
+        /* The X window visual must match the EGL config */
+        XVisualInfo *vis_info, vis_template;
+        int num_visuals;
+        vis_template.visualid = vid;
+        vis_info = XGetVisualInfo (display, VisualIDMask, &vis_template, &num_visuals);
+        if (!vis_info) {
+                g_warning ("Failed to get X visual");
+                goto out;
+        }
+
+        XSetWindowAttributes attr;
+        attr.colormap = XCreateColormap (display, DefaultRootWindow (display),
+                                         vis_info->visual, AllocNone);
+        win = XCreateWindow (display, DefaultRootWindow (display),
+                             0, 0, /* x, y */
+                             1, 1, /* width, height */
+                             0,    /* border_width */
+                             vis_info->depth, InputOutput,
+                             vis_info->visual, CWColormap, &attr);
+        XFree (vis_info);
+
+        eglBindAPI (EGL_OPENGL_ES_API);
+
+        egl_ctx = eglCreateContext (egl_dpy, config, EGL_NO_CONTEXT, ctx_attribs);
+        if (!egl_ctx) {
+                g_warning ("Failed to create EGL context");
+                goto out;
+        }
+
+        egl_surf = eglCreateWindowSurface (egl_dpy, config, win, NULL);
+        if (!egl_surf) {
+                g_warning ("Failed to create EGL surface");
+                goto out;
+        }
+
+        if (!eglMakeCurrent (egl_dpy, egl_surf, egl_surf, egl_ctx)) {
+                g_warning ("eglMakeCurrent() failed");
+                goto out;
+        }
+
+        renderer = g_strdup ((const char *) glGetString (GL_RENDERER));
+
+ out:
+        if (egl_ctx)
+                eglDestroyContext (egl_dpy, egl_ctx);
+        if (egl_surf)
+                eglDestroySurface (egl_dpy, egl_surf);
+        if (egl_inited)
+                eglTerminate (egl_dpy);
+        if (win != None)
+                XDestroyWindow (display, win);
+
+        gdk_error_trap_pop_ignored ();
+        return renderer;
+}
+#endif
+
+static gboolean print_renderer = FALSE;
+
+static const GOptionEntry entries[] = {
+        { "print-renderer", 'p', 0, G_OPTION_ARG_NONE, &print_renderer, "Print EGL renderer name", NULL },
+        { NULL },
+};
+
+int
+main (int argc,
+      char **argv)
+{
+        char *renderer = NULL;
+        GOptionContext *context;
+        int ret = HELPER_NO_ACCEL;
+        GError *error = NULL;
+
+        setlocale (LC_ALL, "");
+
+        context = g_option_context_new (NULL);
+        g_option_context_add_group (context, gtk_get_option_group (TRUE));
+        g_option_context_add_main_entries (context, entries, NULL);
+
+        if (!g_option_context_parse (context, &argc, &argv, &error)) {
+                g_error ("Can't parse command line: %s\n", error->message);
+                g_error_free (error);
+                goto out;
+        }
+
+#ifdef GDK_WINDOWING_X11
+        renderer = get_gles_renderer ();
+#endif
+
+        if (renderer != NULL) {
```

Comment:
please fix cppcheck warning:

`[tools/mate-session-check-accelerated-gles-helper.c:190]: (style) Condition 'renderer!=NULL' is always false`

if `GDK_WINDOWING_X11` is false, the condition `if (renderer != NULL)` is useless code

Reply:
I think that is fixed now.

Reply:
yes, thanks!